### PR TITLE
Read configuration for clickhouse-local from ~/.clickhouse-local

### PR DESCRIPTION
**Pull request by azat**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Read configuration for clickhouse-local from ~/.clickhouse-local

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit 5ea867231bafc01b4512989f351106b7afcc14af with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/67135/5ea867231bafc01b4512989f351106b7afcc14af/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Common/Config/getLocalConfigPath.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        }
+        if (found)
+            break;
+    }
+
+    if (found)
+        return config_path;
+
+    return std::nullopt;
+}
```
Comment (vdimir) [inline src/Common/Config/getLocalConfigPath.cpp:44]: Can't we rewrite it like this?

```suggestion
            if (fs::exists(config_path, ec))
            {
                return config_path;
            }
        }
    }
    return std::nullopt;
}
```

**Review by vdimir — APPROVED** [review]

File: src/Common/Config/getLocalConfigPath.cpp (hunk 1 above)
Comment (azat) [reply src/Common/Config/getLocalConfigPath.cpp:44]: That was just a copy-paste of `getClientConfig`, yep this looks better, applied to both.

**Comment by azat** [comment]
>Runs the unit tests for different release types	

- #66494

**Comment by azat** [comment]
Test failures does not looks related

> Stress test (ubsan) — Found signal in gdb.log

- https://github.com/ClickHouse/ClickHouse/issues/67330

@vdimir does it waits something from my side?

**Comment by vdimir** [comment]
Sorry for pushing it back. Let's merge since failure is unrelated and fixed

**Comment by clickgapai** [comment]
Hi @azat @vdimir — while reviewing this PR I found the following:

- **Bug (P2):** User configurations from `~/.clickhouse-local/config.xml` are silently ignored — https://github.com/ClickHouse/ClickHouse/issues/104932

Happy to discuss — close anything that's wrong or already addressed.

**Pull request merged**